src/python/tmclient/api.py

```diff
@@ -1049,11 +1051,11 @@ def upload_microscope_files(self, plate_name, acquisition_name,
         path = os.path.expandvars(os.path.expanduser(path))
         if os.path.isdir(path):
             filenames = [
-                f for f in os.listdir(directory)
+                f for f in os.listdir(path)
                 if not os.path.isdir(f) and not f.startswith('.')
```

Comment:
I think the following should be used instead

```
                 if not os.path.isdir(os.path.join(path,f)) and not f.startswith('.')
```

f is just the file name, without the path there it is contained. If you want to check if it's a directory you need to append the correct path.

Moreover os.path.isdir() will return false if the file doesn't exist
```
>>> os.path.isdir('/donotexists')
False
```

And this is the reason why you never so an error from it before.

This would probably fix #20 (same code was actually present before this pull request, but since you are there touching it.... :) )